### ISSUE TITLE
[eks] [request]: First class support for nftables kube-proxy backend

### ISSUE THREAD

**Issue by sidewinder12s** [aws/containers-roadmap#2313]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**

First class support for the nftables kube-proxy backend.

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**

The Kubernetes community appears to be moving towards utilizing nftables long term in kube-proxy, over ipvs or iptables modes. Has the EKS Service team started any preliminary discussions about if they will support this mode?

**Are you currently working around this issue?**

Debating if we should move towards the ipvs kube-proxy backend or wait until we can migrate to nftables.

**Additional context**

Beta support is scheduled for 1.31

https://github.com/kubernetes/enhancements/issues/3866

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by stevehipwell** [comment]
It looks like nftables beta support has been [merged](https://github.com/kubernetes/kubernetes/pull/124383) so it's on track for `1.31`. It'd be good to hear what the EKS plans around nftables are prior to the `1.31` release and also if an [eBPF alternative implementation](https://github.com/aws/containers-roadmap/issues/2132) is also being discussed?

**Comment by stevehipwell** [comment]
@mikestef9 could we get an update on this? K8s v1.33 made nftables GA but the EKS [best practices guide](https://docs.aws.amazon.com/eks/latest/best-practices/ipvs.html) still has nftables as under development and recommends IPVS. I'm pretty sure this would "just work" but it'd be good to have confirmation.

**Comment by frittentheke** [comment]
> @mikestef9 could we get an update on this? K8s v1.33 made nftables GA but the EKS [best practices guide](https://docs.aws.amazon.com/eks/latest/best-practices/ipvs.html) still has nftables as under development and recommends IPVS. I'm pretty sure this would "just work" but it'd be good to have confirmation.

And if I may add, IPVS is on its way out: https://github.com/kubernetes/enhancements/issues/5495

**Comment by tkcontiant** [comment]
My question is: Is it going to be features parity between ipvs and nftables?

**Comment by jaydeokar** [comment]
@stevehipwell Yes, it is supported starting v1.31 when nftables was available as a Beta feature in upstream. We will update the best practices doc to reflect this. Thanks for pointing it out
